src/main/java/org/icgc/argo/workflow_management/service/NextflowService.java

```diff
@@ -195,43 +194,38 @@ private CmdKubeRun createCmd(@NonNull Launcher launcher, @NonNull WESRunParams p
-      // Use revision if provided in workflow_engine_options
-      if (nonNull(workflowEngineOptions.getRevision())) {
-        cmdParams.put("revision", workflowEngineOptions.getRevision());
-      }
-
-      // Use launchDir, projectDir and/or workDir if provided in workflow_engine_options
-      if (nonNull(workflowEngineOptions.getLaunchDir())
-          || nonNull(workflowEngineOptions.getProjectDir())
-          || nonNull(workflowEngineOptions.getWorkDir())) {
-        val config =
-            createNextflowConfigFile(
-                runName,
-                workflowEngineOptions.getLaunchDir(),
-                workflowEngineOptions.getProjectDir(),
-                workflowEngineOptions.getWorkDir());
-        cmdParams.put("runConfig", List.of(config));
-      }
+    // Write config file for run (minimum contents == runAsUser)
+    // Use launchDir, projectDir and/or workDir if provided in workflow_engine_options
+    val config =
+        createNextflowConfigFile(
+            runName,
+            k8sConfig.getRunAsUser(),
+            workflowEngineOptions.getLaunchDir(),
+            workflowEngineOptions.getProjectDir(),
+            workflowEngineOptions.getWorkDir());
+    cmdParams.put("runConfig", List.of(config));
+
+    // Resume workflow by name/id
+    if (nonNull(workflowEngineOptions.getResume())) {
+      cmdParams.put("resume", workflowEngineOptions.getResume());
+    }
+    // Use revision if provided in workflow_engine_options
+    if (nonNull(workflowEngineOptions.getRevision())) {
+      cmdParams.put("revision", workflowEngineOptions.getRevision());
+    }
 
-      // should pull latest code before running?
-      // does not prevent us running a specific version (revision),
-      // does enforce pulling of that branch/hash before running)
-      if (nonNull(workflowEngineOptions.getLatest())) {
-        cmdParams.put("latest", workflowEngineOptions.getLatest().equals("true"));
-      }
+    // should pull latest code before running?
+    // does not prevent us running a specific version (revision),
+    // does enforce pulling of that branch/hash before running)
+    if (nonNull(workflowEngineOptions.getLatest())) {
+      cmdParams.put("latest", workflowEngineOptions.getLatest().equals("true"));
+    }
 
-      // Process options (default docker container to run for process if not specified)
-      if (nonNull(workflowEngineOptions.getDefaultContainer())) {
-        val processOptions = new HashMap<String, String>();
-        processOptions.put("container", workflowEngineOptions.getDefaultContainer());
-        cmdParams.put("process", processOptions);
-      }
+    // Process options (default docker container to run for process if not specified)
+    if (nonNull(workflowEngineOptions.getDefaultContainer())) {
+      val processOptions = new HashMap<String, String>();
+      processOptions.put("container", workflowEngineOptions.getDefaultContainer());
+      cmdParams.put("process", processOptions);
```

Comment:
you could do
```java
public static <T> void ifNonNull(@NonNull Consumer<T> consumer, T value){
    if (nonNull(value)){
      consumer.accept(value)
    }

   ifNonNull(workflowEngineOptions.getDefaultContainer(), x -> {
        val processOptions = new HashMap<String, String>();
        processOptions.put("container", x);
        cmdParams.put("process", processOptions);
      });
```


or just keep it like this, since there is only one instance like this
  }